homeassistant/components/lcn.py

```diff
@@ -0,0 +1,168 @@
+                int(matcher.group('id')),
+                is_group)
+        conn_id = matcher.group('conn_id')
+        return addr, conn_id
+    raise vol.error.Invalid('Not a valid address string.')
+
+
+async def async_setup(hass, config):
+    """Set up the LCN component."""
+    import pypck
+    from pypck.connection import PchkConnectionManager
+
+    hass.data[DATA_LCN] = {}
+    hass.data[DATA_LCN][LIB_LCN] = pypck
+
+    conf_connections = config[DOMAIN][CONF_CONNECTIONS]
+    connections = []
+    connection_names = []
+    for conf_connection in conf_connections:
+        # use 'pchk' as default connection_id (or add a numeric suffix if
+        # pchk' is already in use
+        connection_name = conf_connection[CONF_NAME]
+        if connection_name == DEFAULT_NAME:
+            while connection_name in connection_names:
+                suffix = connection_name.strip(DEFAULT_NAME)
+                c_id = 1 if not suffix else int(suffix) + 1
+                connection_name = '{}{:d}'.format(DEFAULT_NAME, c_id)
+        connection_names.append(connection_name)
+
+        settings = {'SK_NUM_TRIES': conf_connection[CONF_SK_NUM_TRIES],
+                    'DIM_MODE': pypck.lcn_defs.OutputPortDimMode[
+                        conf_connection[CONF_DIM_MODE].upper()]}
+
+        connection = PchkConnectionManager(hass.loop,
+                                           conf_connection[CONF_HOST],
+                                           conf_connection[CONF_PORT],
+                                           conf_connection[CONF_USERNAME],
+                                           conf_connection[CONF_PASSWORD],
+                                           settings=settings,
+                                           connection_id=connection_name)
+
+        # establish connection to PCHK server
+        try:
+            await hass.async_create_task(connection.async_connect())
+            connections.append(connection)
+            _LOGGER.info('LCN connected to "{:s}"'.format(connection_name))
+        except TimeoutError:
+            _LOGGER.error('Connection to PCHK server "{:s}" failed.'.format(
+                connection_name))
+            return False
+
+    hass.data[DATA_LCN][CONF_CONNECTIONS] = connections
+    return True
+
+
+class LcnDevice(Entity):
+    """Parent class for all devices associated with the LCN component."""
+
+    def __init__(self, hass, config):
+        """Initialize the LCN device."""
+        self.hass = hass
+        self.pypck = hass.data[DATA_LCN][LIB_LCN]
+        self._name = config[CONF_NAME]
+
+        address, connection_id = config[CONF_ADDRESS]
+        addr = self.pypck.lcn_addr.LcnAddr(*address)
+        connections = hass.data[DATA_LCN][CONF_CONNECTIONS]
+        connection = get_connection(connections, connection_id)
+
+        self.address_connection = connection.get_address_conn(addr)
+        self.address_connection.register_for_inputs(
```

Comment:
What does this do? Update callbacks should be registered in entity coroutine method `async_added_to_hass`.

Reply:
This registers a callback on incoming message objects. Moved to `async_added_to_hass`.